Library/Homebrew/test/dev-cmd/livecheck_spec.rb

```diff
@@ -19,7 +19,19 @@
       .and be_a_success
   end
 
-  it "gives an error when no arguments are given and there's no watchlist", :integration_test do
+  # When a watchlist file is present, livecheck parses the formula/cask names
+  # in the file and ignores any that aren't available. Since formulae and casks
+  # aren't available in the testing environment, the watchlist file will be
+  # treated as empty and this produces an `Invalid usage: No formulae or casks
+  # to check.` error instead. As such, this test will only pass when a
+  # watchlist file isn't present in the testing environment.
+  #
+  # Also note, any older `~/.brew_livecheck_watchlist` file won't be accessible
+  # from this test (as the home folder is `HOMEBREW_PREFIX/Library/Homebrew/
+  # test/`), so we don't have to guard against its existence.
+  it "gives an error when no arguments are given and there's no watchlist",
+     :integration_test,
+     if: !File.exist?(Homebrew::EnvConfig.livecheck_watchlist) do
```

Comment:
Feels like it might be nicer to override the value of this variable in the test? Thoughts?

Reply:
That was my first thought but it felt kind of wrong to be accounting for the presence of a watchlist in a test that expects there to be no watchlist. That said, the following technically works:

```ruby
expected_output = if File.exist?(Homebrew::EnvConfig.livecheck_watchlist)
  /Invalid usage: No formulae or casks to check\./
else
  /Invalid usage: A watchlist file is required when no arguments are given\./
end

expect { brew "livecheck" }
  .to output(expected_output).to_stderr
  .and not_to_output.to_stdout
  .and be_a_failure
```

Reply:
Ah! Now that's an elegant solution. One minute and I'll update this.